node/bin/src/config.rs

```diff
@@ -216,6 +218,16 @@ impl SequencerConfig {
     }
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ConfigAddress(pub Address);
+
+const HASH_DE: Qualified<Serde![str]> =
+    Qualified::new(Serde![str], "hex string with optional 0x prefix");
+impl WellKnown for ConfigAddress {
+    type Deserializer = Qualified<Serde![str]>;
+    const DE: Self::Deserializer = HASH_DE;
+}
```

Comment:
I think you can just bump `smart-config` to `0.4.0` and enable `alloy` feautre. Then you'll just have this "for free"